tools/Tests/JsonUtilitiesTest/JsonExtensionsTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿namespace JsonUtilitiesTest;
+using System;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using Microsoft.Graph.PowerShell.JsonUtilities;
+
+public class JsonExtensionsTests
+{
+    [Fact]
+    public void RemoveDefaultNullProperties_ShouldRemoveDefaultNullValues()
+    {
+        // Arrange
+        JObject json = JObject.Parse(@"{
+            ""displayname"": ""Tim"",
+            ""position"": ""defaultnull"",
+            ""salary"": 2000000,
+            ""team"": ""defaultnull""
+        }");
+
+        // Act
+        string cleanedJson = json.RemoveDefaultNullProperties();
+        JObject result = JObject.Parse(cleanedJson);
+
+        // Assert
+        Assert.False(result.ContainsKey("position"));
+        Assert.False(result.ContainsKey("team"));
+        Assert.Equal("Tim", result["displayname"]?.ToString());
+        Assert.Equal(2000000, result["salary"]?.ToObject<int>());
+    }
+
+    [Fact]
+    public void RemoveDefaultNullProperties_ShouldConvertStringNullToJsonNull()
+    {
+        // Arrange
+        JObject json = JObject.Parse(@"{
+            ""displayname"": ""Tim"",
+            ""position"": ""null"",
+            ""salary"": 2000000,
+            ""team"": """"
+        }");
+
+        // Act
+        string cleanedJson = json.RemoveDefaultNullProperties();
+        JObject result = JObject.Parse(cleanedJson);
+
+        // Assert
+        Assert.Null(result["position"]?.Value<string>());
+        Assert.Null(result["team"]?.Value<string>());
```

Comment:
Curious whether it's correct to be converting explicitly empty strings to null values in the payload.

Reply:
Makes sense. We can remove that conversion for now

Reply:
For Kiota SDKs, we retain the explicit empty strings - https://github.com/microsoft/kiota-dotnet/blob/main/src/serialization/json/JsonSerializationWriter.cs#L93